src/plugins/auto/utils/config.hpp

```diff
@@ -165,7 +167,19 @@ struct PluginConfig {
                 if (!kvp.second.empty())
                     ParsePrioritiesDevices(kvp.second);
                 _devicePriority = kvp.second;
-            } else if (std::find(perf_hints_configs.begin(), perf_hints_configs.end(), kvp.first) != perf_hints_configs.end()) {
+            } else if (kvp.first == ov::hint::execution_mode.name()) {
+                _isSetExecutionMode = true;
+                if (kvp.second == "UNDEFINED") {
+                    _executionMode = 1;
```

Comment:
kvp.second == ov::util::to_string(ov::hint::ExecutionMode::UNDEFINED) is better? How ov::core convert this ov::hint::ExecutionMode::UNDEFINED to string?

Reply:
Yes, it's better to use ov::util::to_string() rather than string "UNDEFINED" directly. Openvino provides the operator '<<' and '>>' overloads function for the property enum value, which could implement the conversion between the enum value and string value.

Reply:
> Why we don't use ov::hint::ExecutionMode::UNDEFINED directly? and _keyConfigMap[ov::hint::execution_mode.name()] = ov::util::to_string(_executionMode)

`UpdateFromMap()` function just check if passed configs are valid and set some config flags, and then `adjustKeyMapValues()` will construct the final config map based on those config flags, which maybe could simplify the `UpdateFromMap()` function. Is it better to update the config map here? or just keep aligned with the original logic of constructing available config map?